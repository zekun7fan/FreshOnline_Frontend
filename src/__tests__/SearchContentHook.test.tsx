import {renderHook, act, cleanup} from '@testing-library/react-hooks';

import SearchContentUI, {SearchContent} from "../components/SearchContent";
import {Provider} from "react-redux";
import store from "../redux/store";
import {BrowserRouter} from "react-router-dom";

jest.mock('antd/es/tree-select', () => ({}));

afterEach(() => {
    cleanup().then()
})

test('render search content using hook', () => {
    const wrapper = ({ children }:{ children: any }) => (
        <Provider store={store}>
            <BrowserRouter>
                {children}
            </BrowserRouter>
        </Provider>
    )
    const { result } = renderHook(() => SearchContent(), { wrapper })
    const { search_results, onChange } = result.current
    act(() => onChange(1))
    expect(search_results).toEqual([])
})

test('render search content UI using hook', () => {
    const wrapper = ({ children }:{ children: any }) => (
        <Provider store={store}>
            <BrowserRouter>
                {children}
            </BrowserRouter>
        </Provider>
    )
    const { result } = renderHook(() => SearchContentUI(), { wrapper })
    expect(result.current).toMatchSnapshot()
})