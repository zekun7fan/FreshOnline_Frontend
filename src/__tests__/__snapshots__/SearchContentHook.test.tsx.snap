// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render search content UI using hook 1`] = `
<Layout>
  <Row>
    <div>
      No Goods
    </div>
  </Row>
  <Layout>
    <Row
      align="middle"
      justify="center"
    >
      <Col>
        <Pagination
          defaultCurrent={1}
          onChange={[Function]}
          pageSize={20}
          total={0}
        />
      </Col>
    </Row>
  </Layout>
</Layout>
`;
